src/back/TicketManager/src/Core/TicketManager.Core.Domain/Events/Event.cs

```diff
@@ -23,5 +24,66 @@ public Event(Guid organizerId, string name, string description, string location,
         Description = description;
         Location = location;
         Date = date;
+        Status = EventStatus.Verified;
+    }
+
+    public void ChangeEventStatus(EventStatus status)
+    {
+        switch(status)
+        {
+            case EventStatus.Verified:
+                if(Status != EventStatus.Unverified &&  Status != EventStatus.Cancelled)
+                {
+                    throw new InvalidOperationException("Cannot change event status to Verified from current status.");
+                }
+                break;
+            case EventStatus.Published:
+                if(Status != EventStatus.Verified)
+                {
+                    throw new InvalidOperationException("Cannot change event status to Published from current status.");
+                }
+                break;
+            case EventStatus.Opened:
+                if (Status != EventStatus.Published && Status != EventStatus.Closed)
+                {
+                    throw new InvalidOperationException("Cannot change event status to Opened from current status.");
+                }
+                break;
+            case EventStatus.Closed:
+                if (Status != EventStatus.Opened)
+                {
+                    throw new InvalidOperationException("Cannot change event status to Closed from current status.");
+                }
+                break;
+            case EventStatus.Finished:
+                if (Status != EventStatus.Opened && Status != EventStatus.Closed)
+                {
+                    throw new InvalidOperationException("Cannot change event status to Finished from current status.");
+                }
+                break;
+            case EventStatus.Cancelled:
+                if(Status != EventStatus.Verified && Status != EventStatus.Published)
+                {
+                    throw new InvalidOperationException("Cannot change event status to Cancelled from current status.");
+                }
+                break;
+            case EventStatus.Held:
+                if(Status != EventStatus.Published && Status != EventStatus.Opened)
+                {
+                    throw new InvalidOperationException("Cannot change event status to Held from current status.");
+                }
+                break;
+            case EventStatus.Recalled:
+                if(Status != EventStatus.Held && Status != EventStatus.Opened)
+                {
+                    throw new InvalidOperationException("Cannot change event status to Recalled from current status.");
+                }
+                break;
+            case EventStatus.Unverified:
+            default:
+                throw new InvalidOperationException("Cannot change event status to the specified value.");
+        }
+
+        Status = status;
```

Comment:
You can divide this monster into many smaller methods like `Verify()`, `Close()` or `Finish()`.